### ISSUE TITLE
A false negative about the rule NP_NULL_ON_SOME_PATH

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#2030]
Hi, I found a interesting false negative about the rule [NP_NULL_ON_SOME_PATH](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#np-possible-null-pointer-dereference-np-null-on-some-path). Please read the following code example:
```java
public static void test(Object o) {
    for (int i = 0; i < 1; i++) {
      if (o == null)
        new IllegalArgumentException("Forgot to throw this");
    }
    if (o instanceof String) {
        if (o == null)
            System.out.println("This check is redundant");
        return;
    }
    System.out.println(o.hashCode());
}
```
Suppose we remove the for loop, like the following case, SpotBugs can report a warning about this rule:
```java
public static void test(Object o) {
    if (o == null)  // After removing for loop
      new IllegalArgumentException("Forgot to throw this");
    if (o instanceof String) {
        if (o == null)
            System.out.println("This check is redundant");
        return;
    }
    System.out.println(o.hashCode());
}
```
These two cases are equivalent, hence I think this is a false negative. Thanks for your consideration.

SpotBugs Version: Latest

**Comment by hzy20001211** [comment]
I don't understand your question. Your two sample code both show [NP_NULL_ON_SOME_PATH] in line "println" in my test. Would you mind show more details about your question?
